ui/lib/utils/routing.ts

```diff
@@ -10,6 +10,10 @@ export function isLocationMatch(s, matchPrefix) {
   }
 }
 
-export function isLocationMatchPrefix(s) {
+export function isLocationMatchPrefix(s): boolean {
   return isLocationMatch(s, true)
 }
+
+export function isPortalPage(): boolean {
+  return window.location.pathname.endsWith('/portal')
```

Comment:
I realized a problem. Since we support customizing publicPath, `/portal` is a valid publicPath that may be specified by user. Maybe we must put it after `#`.

Reply:
Yes, it is, put it in the hash is better. Done!